.github/workflows/build.yml

```diff
@@ -39,8 +39,23 @@ jobs:
         registry: ${{ env.REGISTRY }}
         username: ${{ github.actor }}
         password: ${{ secrets.GITHUB_TOKEN }}
-    - name: Build and test Docker image
-      id: build
+    - name: Build test environment Docker image
+      uses: docker/build-push-action@v2
+      with:
+        context: .
+        target: test
+        tags: ${{ env.TEST_TAG }}
+        load: true
+        cache-from: type=gha
+        cache-to: type=gha,mode=max # mode=max means cache intermediate images
+        build-args: |
+          GIT_REVISION=${{ github.sha }}
+          CI=true
+          GITHUB_ACTIONS=true
+    - name: Run tests
+      run: |
+        docker run -e PATH_PREFIX=${{ github.workspace }} -e CI=true -e GITHUB_ACTIONS=true --rm ${{ env.TEST_TAG }}
```

Comment:
`env.TEST_TAG` doesn't appear to be set, which is why the actions run failed when it got to here and docker run had no actual container image argument passed to it

Reply:
Yeah looks like the cherry-pick didn't work the way I thought it would. I'll clean this up so that it actually has all of the changes from #674.